ocaml/xapi/helpers.ml

```diff
@@ -2360,3 +2360,16 @@ module AuthenticationCache = struct
           None
   end
 end
+
+let assert_certificates_exchanged_in_pool ~__context =
+  if not (Sys.file_exists !Xapi_globs.trusted_pool_certs_dir) then
+    raise Api_errors.(Server_error (host_certificates_not_exchanged, [])) ;
+
+  let num_cert_files =
+    Sys.readdir !Xapi_globs.trusted_pool_certs_dir
```

Comment:
Hmm... I thought we were using a bundle file rather than a dir...

Reply:
These are the PEM files on an XS9 system. Not all are from us

```
/etc/stunnel/certs-pool/017e419f-e9a9-4724-856e-5547358de828.pem
/etc/stunnel/xapi-stunnel-ca-bundle.pem
/etc/stunnel/xapi-pool-ca-bundle.pem
/etc/xensource/xapi-pool-tls.pem
/etc/xensource/xapi-ssl.pem
/etc/unbound/icannbundle.pem
/etc/pki/ca-trust/extracted/pem/objsign-ca-bundle.pem
/etc/pki/ca-trust/extracted/pem/email-ca-bundle.pem
/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
/etc/pki/tls/cert.pem
```

Reply:
Stunnel is configured to use the bundles as its trust root: https://github.com/xapi-project/xen-api/blob/master/ocaml/libs/stunnel/stunnel.ml#L237